.github/workflows/rill-cloud.yml

```diff
@@ -0,0 +1,52 @@
+# This workflow triggers deployment with Rill Cloud helm charts
+# Each merge to main branch is build with github sha tag and published to Rill Cloud.
+name: Deploy to Rill Cloud
+on:
+  push:
+    tags:
+      - "**"
+    branches:
+      - main
+      - cd
+  workflow_dispatch:
+
+env:
+  RELEASE: ${{ contains('refs/tags', github.ref_name) }}
+
+jobs:
+  release:
+    name: Deploy to Rill Cloud
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19.2
+
+      - name: Set up GCloud SDK
+        uses: google-github-actions/setup-gcloud@v1
+
+      - name: Authenticate GCloud
+        uses: google-github-actions/auth@v1
+        with:
+          credentials_json: "${{ secrets.RILL_BINARY_SA }}"
+
+      - name: Build & Publish Rill docker image
+        run: |-
+          go build -o rill cli/main.go
+
+          gcloud auth configure-docker
+
+          docker build -t gcr.io/rilldata/rill-headless:${GITHUB_SHA} .
+          docker push gcr.io/rilldata/rill-headless:${GITHUB_SHA}
+
+      - name: Trigger Rill Cloud deployment
+        run: |-
+          set -e
+          curl -X POST https://api.github.com/repos/rilldata/rill-helm-charts/dispatches \
+          -H "Accept: application/vnd.github.everest-preview+json" \
+          -H "Authorization: token ${{ secrets.GORELEASER_ACCESS_TOKEN }}" \
+          --data '{"event_type": "Deploying Tag: '"${GITHUB_REF_NAME}"'", "client_payload": { "github_sha": "'"${GITHUB_SHA}"'", "tag": "'"${GITHUB_REF_NAME}"'", "release": "${{ env.RELEASE }}"}}'
```

Comment:
Should we remove the tag and release info – not sure it makes sense here (e.g. will be CD from main, and the release versions of local won't really relate to deployed versions)

Reply:
we need this info at our helm charts so that we can update the image tag to sha or tag depending on the release or stage.